Ben987/Bondage-College
# Refactor: Merge round and weighted nipple piercings

# Summary
As both, the round piercing and the weighted nipple piercings share the same graphic representation, I merged both items, when I migrated the round nipple piercing to the extended item script. Thus there are now four options, after you pierced the nipples:
* Base
* Tied to the collar
* Weighted
* Weighted and tied to the collar
<img width="379" alt="image" src="https://user-images.githubusercontent.com/61221552/91636199-15ce5f00-e9ff-11ea-96d9-17205494f975.png">
This PR relies on the extended items extension #1326  by @Nina-1474 

# Details

## Changed files
No new files were added, but this PR
* copied all weighted piercing images to the `RoundPiercing` directory and changed the file names appropriately
* removed the weighted piercing directory
* removed the weighted piercing script from `index.html'
* changed `RoundPiercing.js` to use the extended item script
* added a `Type` attribute to the `RemoveItemOnRemoval` asset, so that extended items of a special type now also are removed automatically, when the base item is removed
* removed entries for the WeightedPiercing from `Female3DCG.js` and `Female3DCG.csv` and raising the price for the round piercing to 40$, because now the set comes with weights

## Testing
I tested the changes locally. The following test cases were done
* Check the graphics for all upper body sizes
* Check, if locks can be applied to all variants
* Check for prerequisites
* Apply the item with all variants to
  * The player
  * Another player
  * An NPC
* Check for the right action messages, if the item is used in a chat room

## Credits
Go to @elliesec for helping me with the base concept, @Nina-1474 for the extended item PR and @ace-1331 for quality assurance
